L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc

```diff
@@ -0,0 +1,133 @@
+#include "L1Trigger/L1TMuonEndCap/interface/TTPrimitiveConversion.h"
+
+#include "L1Trigger/L1TMuonEndCap/interface/SectorProcessorLUT.h"
+#include "L1Trigger/L1TMuonEndCap/interface/TrackTools.h"
+
+
+void TTPrimitiveConversion::configure(
+    const TTGeometryTranslator* tp_ttgeom,
+    const SectorProcessorLUT* lut,
+    int verbose, int endcap, int sector, int bx
+) {
+  assert(tp_ttgeom != nullptr);
+  assert(lut != nullptr);
+
+  tp_ttgeom_ = tp_ttgeom;
+  lut_       = lut;  // not used
+
+  verbose_ = verbose;
+  endcap_  = endcap; // 1 for ME+, 2 for ME-
+  sector_  = sector;
+  bx_      = bx;
+}
+
+void TTPrimitiveConversion::process(
+    const std::map<int, TTTriggerPrimitiveCollection>& selected_ttprim_map,
+    EMTFHitCollection& conv_hits
+) const {
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_it  = selected_ttprim_map.begin();
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_end = selected_ttprim_map.end();
+
+  for (; map_tp_it != map_tp_end; ++map_tp_it) {
+    TTTriggerPrimitiveCollection::const_iterator tp_it  = map_tp_it->second.begin();
+    TTTriggerPrimitiveCollection::const_iterator tp_end = map_tp_it->second.end();
+
+    for (; tp_it != tp_end; ++tp_it) {
+      EMTFHit conv_hit;
+      convert_tt(*tp_it, conv_hit);
+      conv_hits.push_back(conv_hit);
+    }
+  }
+}
+
+void TTPrimitiveConversion::process_no_prim_sel(
+    const TTTriggerPrimitiveCollection& ttmuon_primitives,
+    EMTFHitCollection& conv_hits
+) const {
+  TTTriggerPrimitiveCollection::const_iterator tp_it  = ttmuon_primitives.begin();
+  TTTriggerPrimitiveCollection::const_iterator tp_end = ttmuon_primitives.end();
+
+  for (; tp_it != tp_end; ++tp_it) {
```

Comment:
Please make this auto-range for loop with
`for (auto& tp : TTTriggerPrimitiveCollection)`

No need for `tp_it` and `tp_end` above.